<a href="https://colab.research.google.com/github/SIESOBOT/DS-Unit-1-Sprint-1-Data-Wrangling-and-Storytelling/blob/master/DS-Unit-1-Sprint-2-Statistics/module1/DS13_TLBTR_LUISFELIPE_URENA_2_1_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<img align="left" src="https://lever-client-logos.s3.amazonaws.com/864372b1-534c-480e-acd5-9711f850815c-1524247202159.png" width=200>
<br></br>
<br></br>

## *Data Science Unit 1 Sprint 2 Assignment 1*

# Apply the t-test to real data

Your assignment is to determine which issues have "statistically significant" differences between political parties in this [1980s congressional voting data](https://archive.ics.uci.edu/ml/datasets/Congressional+Voting+Records). The data consists of 435 instances (one for each congressperson), a class (democrat or republican), and 16 binary attributes (yes or no for voting for or against certain issues). Be aware - there are missing values!

Your goals:

1. Load and clean the data (or determine the best method to drop observations when running tests)
2. Using hypothesis testing, find an issue that democrats support more than republicans with p < 0.01
3. Using hypothesis testing, find an issue that republicans support more than democrats with p < 0.01
4. Using hypothesis testing, find an issue where the difference between republicans and democrats has p > 0.1 (i.e. there may not be much of a difference)

Note that this data will involve *2 sample* t-tests, because you're comparing averages across two groups (republicans and democrats) rather than a single group against a null hypothesis.

Stretch goals:

1. Refactor your code into functions so it's easy to rerun with arbitrary variables
2. Apply hypothesis testing to your personal project data (for the purposes of this notebook you can type a summary of the hypothesis you formed and tested)

In [0]:
### YOUR CODE STARTS HERE
import pandas as pd
import numpy as np
from scipy import stats

In [0]:
df = pd.read_csv('https://archive.ics.uci.edu/ml/machine-learning-databases/voting-records/house-votes-84.data',names=['class_name','handicapped_infants','water_proyect_cost_sharing',
                                                                                                                       'adoption_of_the_budget_resolution','phicisian_fee_freeze',
                                                                                                                       'el_salvador_aid','religious_groups_in_schools','anti_satellite_test_ban',
                                                                                                                       'aid_to_nicaraguan_contras','mx_missile','immigration','synfuels_corporation_cutback',
                                                                                                                       'education_spending','superfund_right_to_sue','crime','duty_free_exports','export_administration_act_south_africa'])

In [29]:
df.head(10)

class_name  ... export_administration_act_south_africa
0  republican  ...                                      y
1  republican  ...                                      ?
2    democrat  ...                                      n
3    democrat  ...                                      y
4    democrat  ...                                      y
5    democrat  ...                                      y
6    democrat  ...                                      y
7  republican  ...                                      y
8  republican  ...                                      y
9    democrat  ...                                      ?

[10 rows x 17 columns]

In [0]:
df.replace({'y':1,'n':0,'?':np.nan},inplace=True)

In [31]:
dem = df[df['class_name']=='democrat']

dem.head()
print(dem.shape)

(267, 17)


In [32]:
rep = df[df['class_name']=='republican']
print(rep.shape)
rep.head()

(168, 17)


class_name  ...  export_administration_act_south_africa
0   republican  ...                                     1.0
1   republican  ...                                     NaN
7   republican  ...                                     1.0
8   republican  ...                                     1.0
10  republican  ...                                     0.0

[5 rows x 17 columns]

# test 1
**Null Hipothesis**: The democrats and republicans support the aid to el salvador equally.
**Alternative Hipothesis**: The democrats and republicans do not support aid 
to el savador equally.

**confidence level**: 95% (1-.95= .05)



In [33]:
stats.ttest_ind(dem['el_salvador_aid'],rep['el_salvador_aid'], nan_policy='omit')

Ttest_indResult(statistic=-21.13669261173219, pvalue=5.600520111729011e-68)

by the results of the t test, with a ttest of -21.13 and a p value of 5.60, i reject the null hipothesis, and suggest the alternative: the democrats and
republicans do not support this bill equally**(republicans supported the bill more than the democrats**)

# TEST 2

**Null Hipothesis:** The democrats and republicans support the adoption of the budget resolution bill equally.
**Alternative Hipothesis** the democrats and republicans *dont* support this bill
equally.
**confidence level:** 95%

In [34]:
stats.ttest_ind(dem['adoption_of_the_budget_resolution'],rep['adoption_of_the_budget_resolution'],nan_policy='omit')

Ttest_indResult(statistic=23.21277691701378, pvalue=2.0703402795404463e-77)

**Conclusion:** being the result of the Ttest t value = 23.21 and p value = 2.07
i reject the null hipothesis and suggest that the alternative hipothesis, is the most likely (*democrats supported the bill more than the republicans*)

# TEST 3
**Null Hipothesis:** republicans and democrats support the bill immigration equally.
**alternate hipothesis:** republicans and democrats do not support the bill
equally.
**confidence level**: 95%

In [46]:
stats.ttest_ind(dem['immigration'],rep['immigration'],nan_policy='omit')

Ttest_indResult(statistic=-1.7359117329695164, pvalue=0.08330248490425066)

**Conclusion:** by the results of the test, tvalue = -1.73 and pvalue = 0.08
i accept the null hipothesis that ***both parties support this bill equally***



---

# STRETCH GOAL

In [0]:
#defining a function to make the code resusable
def ttest(tval,pval):
 result = stats.ttest_ind(tval,pval,nan_policy='omit')
 return result

In [52]:
ttest(dem['immigration'],rep['immigration'])

Ttest_indResult(statistic=-1.7359117329695164, pvalue=0.08330248490425066)